### ISSUE TITLE
Add MaxItemsToReturn to the request model to allow specifying values other than 100

### ISSUE THREAD

**Issue by lynkz-matt-psaltis** [OmniSharp/omnisharp-roslyn#2513]
https://github.com/OmniSharp/omnisharp-roslyn/blob/7299d0836af701b61dbbeb823c6a1efbb3010867/src/OmniSharp.LanguageServerProtocol/Handlers/OmniSharpWorkspaceSymbolsHandler.cs#L49

Pretty please?

**Comment by filipw** [comment]
This is already exposed as a custom property on the OmniSharp's own custom API.
However in the LSP server it's not defined as the property by the spec (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspaceSymbolParams).

**Comment by matt-psaltis** [comment]
Thanks @filipw that makes a heap of sense! I was wondering, from the spec it doesn't seem to specify a behaviour of 100 items as a limit but instead says to use a blank query string to return all items. Does that make the current behaviour non-compliant? Cheers!

**Comment by filipw** [comment]
yes this is an arbitrary value chose to not hinder performance. At the same time, as you provide more characters it then should narrow down.

What we can do - I guess - is to expose it in the OmniSharp config, where you can tweak such "defaults" for yourself?
This was previously brought up for similar related topics, e.g. request timeouts
